lgc/state/PipelineState.cpp

```diff
@@ -700,31 +728,19 @@ std::pair<const ResourceNode *, const ResourceNode *>
 PipelineState::findResourceNode(ResourceNodeType nodeType, unsigned descSet, unsigned binding) const {
   for (const ResourceNode &node : getUserDataNodes()) {
     if (node.type == ResourceNodeType::DescriptorTableVaPtr) {
-      if (nodeType == node.type) {
+      if (nodeType == ResourceNodeType::DescriptorTableVaPtr) {
         if (!node.innerTable.empty() && node.innerTable[0].set == descSet)
```

Comment:
I understand your hesitation to change too much at once, but would you consider removing the node.innerTable.empty() check as a followup? That check seems not to have a good justification.

Reply:
Why is this check not needed?  Is the inner table guaranteed to have at least one element?  If not, then accessing element 0 in the inner table could cause undefined behaviour.

Reply:
I turned this check into an assertions. I compiled all pipelines from an UE4 benchmarks and didn't fail. But I am too interested in why this must be the case.

Reply:
Part of the confusion may be something else that I missed: Why is the comparison `node.innerTable[0].set == descSet`, and not simply `node.set == descSet`? That seems more intuitive to me.